vms/types/blob_data.go

```diff
@@ -9,13 +9,33 @@ import (
 	"github.com/ava-labs/avalanchego/utils/formatting"
 )
 
+const nullStr = "null"
+
 // JSONByteSlice represents [[]byte] that is json marshalled to hex
 type JSONByteSlice []byte
 
 func (b JSONByteSlice) MarshalJSON() ([]byte, error) {
+	if b == nil {
+		return []byte(nullStr), nil
+	}
+
 	hexData, err := formatting.Encode(formatting.HexNC, b)
 	if err != nil {
 		return nil, err
 	}
 	return json.Marshal(hexData)
 }
+
+func (b *JSONByteSlice) UnmarshalJSON(jsonBytes []byte) error {
+	if string(jsonBytes) == nullStr {
+		return nil
+	}
+
+	var hexData string
+	err := json.Unmarshal(jsonBytes, &hexData)
+	if err != nil {
+		return err
+	}
+	*b, err = formatting.Decode(formatting.HexNC, hexData)
```

Comment:
(optional) I typically try to ensure that error paths don't leak other values, but appreciate that the behaviour is undefined so there's no clear-cut "correct" answer:

```go
v, err := formatting.Decode(...)
if err != nil {
    return err
}
*b = v
return nil
```